core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java

```diff
@@ -245,16 +282,20 @@ public void setRetryPolicy(RetryPolicy retryPolicy) {
             retryPolicy(retryPolicy);
         }
 
+        @Override
+        public RetryPolicy retryPolicy() {
+            return retryPolicy;
+        }
+
         @Override
         public Builder executionInterceptors(List<ExecutionInterceptor> executionInterceptors) {
-            this.executionInterceptors.clear();
-            this.executionInterceptors.addAll(executionInterceptors);
+            this.executionInterceptors = Collections.unmodifiableList(executionInterceptors);
```

Comment:
I don't believe this gives us the guarentees we want here. Any modification to the original list will still be respected, you just can't modify the unmodifiable view you create.